src/app/features/course-info/course-info.component.ts

```diff
@@ -1,19 +1,47 @@
-import { Component, EventEmitter, Input, Output } from "@angular/core";
-import { ButtonTypes } from "@app/shared/types/button.type";
-import { CourseInfo } from "@app/shared/types/course.model";
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { MappingService } from '@app/services/mapping.service';
+import { ButtonTypes } from '@app/shared/types/button.type';
+import { CourseInfo } from '@app/shared/types/course.model';
+import { combineLatest, Subscription } from 'rxjs';
 
 @Component({
-  selector: "app-course-info",
-  templateUrl: "./course-info.component.html",
-  styleUrls: ["./course-info.component.scss"],
+  selector: 'app-course-info',
+  templateUrl: './course-info.component.html',
+  styleUrls: ['./course-info.component.scss'],
 })
-export class CourseInfoComponent {
+export class CourseInfoComponent implements OnInit, OnDestroy {
   ButtonTypes = ButtonTypes;
+  course: CourseInfo | undefined = { id: '', title: '', duration: 0, description: '', authors: [], date: '' };
+  courseId: string | undefined;
```

Comment:
It is not necessary to create a class variable for the ID as the course object itself has it.
This is only used in the subscription where it is safe to operate with fn scope variable.

Reply:
yes right! Done